### ISSUE TITLE
DOC: Dataframe.skew invalid function signature for "skipna"

### ISSUE THREAD

**Issue by bf** [pandas-dev/pandas#34063]
#### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.skew.html

#### Documentation problem

In the function signature it says `skipna=None` even though the parameter documentation underneath says `skipna bool, default True`.

#### Suggested fix for documentation

The `skipna=None` in function signature should be changed to `skipna=True`.

**Comment by MarcoGorelli** [comment]
Thanks @bf - are you interested in opening a pull request?

**Comment by bf** [comment]
@MarcoGorelli I'd love to create a PR, but I have trouble finding the proper source file. 

Could you help me find the root cause? 

I have found https://github.com/pandas-dev/pandas/blob/dec736f3fcc3337bea0ee180ee28f73e5bf54939/pandas/core/generic.py#L10197 which seems to indicate that the definition of `_make_stat_function()` in https://github.com/pandas-dev/pandas/blob/dec736f3fcc3337bea0ee180ee28f73e5bf54939/pandas/core/generic.py#L11139 erroneously sets `skipna=None` at https://github.com/pandas-dev/pandas/blob/dec736f3fcc3337bea0ee180ee28f73e5bf54939/pandas/core/generic.py#L11161

But as `_make_stat_function()` is used all over the place in that file, we'd need to check every single function if the default `skipna=True` is really correct.

Do you agree with this process?

**Comment by MarcoGorelli** [comment]
Thanks for taking a look

Have just had a look at the source code for `_make_stat_function`, and it says
```python
        if skipna is None:
            skipna = True
```

TBH I don't understand why the default value is set to `None`, and then it's changed to `True` in every case.

@simonjayhawkins would it be OK to change `_make_stat_function`'s signature to have
```python
skipna=True
```
and then remove the lines
```python
        if skipna is None:
            skipna = True
```
so that in the documentation, the default value of True (from `_num_doc`) would match the default shown in the signature?

**Comment by simonjayhawkins** [comment]
@bf Thanks for the report.

the latest docs are at https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.skew.html

The type annotations in the function signature are now removed to avoid this type of conflict with the paramater descriptions, see #33312

**Comment by bf** [comment]
You are right, the issue exists also for other doc pages which use `_make_stat_function`, e.g. https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.std.html

I imagine the `if skipna is None: skipna = True` is most likely done to enable some static type checking for the underlying functions, or to remove some weird testing error, as it makes sure all functions only receive a proper boolean parameter.

**Comment by bf** [comment]
@simonjayhawkins seems like we posted at the same time. Thanks for the fix and for your great work.

**Comment by simonjayhawkins** [comment]
I'll reopen, since even without the type annotations, the default is still shown as None

**Comment by MarcoGorelli** [comment]
@simonjayhawkins isn't this still present even without the type annotations?
![image](https://user-images.githubusercontent.com/33491632/81402149-c353f480-9128-11ea-97ad-4960478ffe32.png)

**Comment by bf** [comment]
Not sure if I need to create a new issue, but in the docs for `Dataframe.mode` the skipna parameter is suddenly called `dropna`. https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.DataFrame.mode.html

Maybe it makes sense to either call everything `skipna` or `dropna`.

**Comment by MarcoGorelli** [comment]
@bf that would be a separate issue
